### DISCUSSION TITLE
annotateGiotto() function + celldex reference file for single cell analysis = Annotation Interrupted

### DISCUSSION THREAD

**Discussion by tsoruu** [giotto-suite/Giotto#755] [Q&A]
Hi, I have a problem with running the annotateGiotto() function with a reference file for single-cell analysis from the celldex library. For some background, I'm trying to merge the tutorial for the Mouse Brain Visium Tutorial with the Single Cell Analysis tutorial and annotate it with the Celldex reference. The spatial data used are the same from the tutorial while for the Single Cell Analysis, I use [10X Genomics 5k Adult Mouse Brain Nuclei](https://www.10xgenomics.com/resources/datasets/5k-adult-mouse-brain-nuclei-isolated-with-chromium-nuclei-isolation-kit-3-1-standard) and for the reference file, I use the [Mouse RNA-seq data from celldex library](https://bioconductor.org/packages/devel/data/experiment/vignettes/celldex/inst/doc/userguide.html#23_Mouse_RNA-seq). The figure below shows the line of code, as well as the output directly after it was run. 

![image](https://github.com/drieslab/Giotto/assets/83799574/e51a527d-78ae-43b4-9dd9-55e8b3a92856)

While this is the last few lines of the output's error line.

![image](https://github.com/drieslab/Giotto/assets/83799574/36f2fd50-d38c-4ef0-9958-32d82693086e)

I'd like to know:
- What is the intended format of the input file for the 'annotation_vector' parameter for the annotateGiotto() function?
- If the annotateGiotto() function could be used with the celldex reference, how should I format/process the reference file?
- If the annotateGiotto() function couldn't be used with the celldex reference, are there any alternative tools that I could use to annotate the with? or is there any other reference database/set that I might use as a substitute for the celldex reference?

Please teach me how to do the proper way to start a discussion as I'm not well informed on whether/how to share my log file of the function if this discussion is lacking in that department. Sorry for such inconvenience that I might cause. I hope that this problem is quickly resolved and wish the Giotto Suite team all the best in developing these tools.

**Comment by tsoruu** [comment]
scRNA-seq data have to be processed using SingleR with one of the celldex references to predict the genes of each cluster. Manual creation of a data frame with the name of the genes and which cluster that it has to be assigned needed to be done after the prediction to utilize the annotateGiotto() function. Sorry for this post filling up the discussion page... Should've RTFM...

**Reply by RubD** [reply]
Seems like we missed this discussion thread, but I believe you have figured it out yourself, correct?
